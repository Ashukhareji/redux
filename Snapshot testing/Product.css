@media (max-width: 800px) {
  div {
    display: flex;
    flex-direction: column;

  }
}

.Product {
  display: flex;
  justify-content: center;
  margin: 10px;
  background-color: beige;
  height: 400px;
  align-items: center;
}