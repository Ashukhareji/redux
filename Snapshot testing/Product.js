import React from 'react'
import './Product.css'
const Product = ({ items = [] }) => {
    if (items.length <= 0) {
        return <div>No items Found</div>
    }

    else {
        return <div>
            {items.map((items, index) => {
                return <h4 key={index} >{items}</h4>
            })}
        </div>
    }
}


export default Product