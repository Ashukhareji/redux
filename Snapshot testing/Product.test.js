import Product from './Product';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer
    .create(<Product/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('render when there are no items', () =>{
    const snapshopt = renderer.create(<Product/>).toJSON();
    expect(snapshopt).toMatchSnapshot();
});
it('render when there is one item', () =>{
    const snapshopt = renderer.create(<Product items={['Rice']}/>).toJSON();
    expect(snapshopt).toMatchSnapshot();
});
it('render when there are multiple items', () =>{
    const snapshopt = renderer.create(<Product items={["Rice","Pulses","Wheat"]}/>).toJSON();
    expect(snapshopt).toMatchSnapshot();
});
