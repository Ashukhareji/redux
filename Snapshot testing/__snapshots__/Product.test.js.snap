// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render when there are multiple items 1`] = `
<div>
  <h4>
    Rice
  </h4>
  <h4>
    Pulses
  </h4>
  <h4>
    Wheat
  </h4>
</div>
`;

exports[`render when there are no items 1`] = `
<div>
  No items Found
</div>
`;

exports[`render when there is one item 1`] = `
<div>
  <h4>
    Rice
  </h4>
</div>
`;

exports[`renders correctly 1`] = `
<div>
  No items Found
</div>
`;
